pkg/trtl/honu.go

```diff
@@ -27,6 +28,10 @@ func NewHonuService(s *Server) (*HonuService, error) {
 func (h *HonuService) Get(ctx context.Context, in *pb.GetRequest) (*pb.GetReply, error) {
 	var err error
 
+	if _, found := reservedNamespaces[string(in.Namespace)]; found {
+		log.Error().Msg("cannot use reserved namespace")
+		return nil, status.Error(codes.PermissionDenied, "cannot use reserved namespace")
+	}
```

Comment:
For ease of managing logging errors, I'd recommend that we `log.Warn()` here rather than error - the user has supplied a bad request, but can change the namespace and try again with no action on our part, whereas errors are more severe alerts that require our attention.

Also, the `codes.PermissionDenied` was an interesting choice, and it makes sense. Generally I think of `PermissionDenied` as having a specific user context, e.g. user A is permission denied but user B is not, but I can see how this would make sense, permission is denied for all users to access this resource and the same request should not be tried again. What would you think about using `INVALID_ARGUMENT`, e.g. you can retry as long as you use a different namespace?

Reply:
Gotcha - it looked (below) like we were erring on the side of erroring rather than warning, so just trying to be consistent with what was already there. `PermissionDenied` was meant in the sense of "you do not have access to this namespace"

Reply:
That's a good point, we should also change the errors below to `log.Warn()`. @pdeziel did we discuss that in the `Get` PR? I thought we had talked about not making `Options` required since it is a better user experience to provide `nil` rather than an empty object? 

Ok, I'm fine with sticking to `PermissionDenied` if you think that's what it should be.

Reply:
I remember seeing your comment about that but I think I missed it in the PR. I agree it's a better user experience to provide `nil` if you just want the value.

Reply:
Trying to pick up the thread from above @bbengfort @pdeziel -- does that mean you guys want me to remove:

```golang
if in.Options == nil {
	log.Error().Msg("missing options in Trtl Get request")
	return nil, status.Error(codes.InvalidArgument, "options must be provided in Get request")
}
```

or change to Warn and `return nil, nil`?